proposals/NNNN-observability.md

```diff
@@ -0,0 +1,760 @@
+
+    init(order: Order? = nil, account: Account? = nil, alternateIconsUnlocked: Bool = false, allRecipesUnlocked: Bool = false) {
+        _$observationStorage = _$ObservationStorage(order: order, account: account, alternateIconsUnlocked: alternateIconsUnlocked, allRecipesUnlocked: allRecipesUnlocked)
+    }
+  
+    var order: Order? {
+        get { 
+            _$observationRegistrar.access(self, keyPath: \.order)
+            return _$observationStorage.order
+        }
+        set {
+            _$observationRegistrar.withMutation(self, keyPath: \.order) {
+                _$observationStorage.order = newValue
+            }
+        }
+    }
+  
+    var account: Account? {
+        get { 
+            _$observationRegistrar.access(self, keyPath: \.account)
+            return _$observationStorage.account
+        }
+        set {
+            _$observationRegistrar.withMutation(self, keyPath: \.account) {
+                _$observationStorage.account = newValue
+            }
+        }
+    }
+
+    var alternateIconsUnlocked: Bool {
+        get { 
+            _$observationRegistrar.access(self, keyPath: \.alternateIconsUnlocked)
+            return _$observationStorage.alternateIconsUnlocked
+        }
+        set {
+            _$observationRegistrar.withMutation(self, keyPath: \.alternateIconsUnlocked) {
+                _$observationStorage.alternateIconsUnlocked = newValue
+            }
+        }
+    }
+
+    var allRecipesUnlocked: Bool {
+        get { 
+            _$observationRegistrar.access(self, keyPath: \.allRecipesUnlocked)
+            return _$observationStorage.allRecipesUnlocked
+        }
+        set {
+            _$observationRegistrar.withMutation(self, keyPath: \.allRecipesUnlocked) {
+                _$observationStorage.allRecipesUnlocked = newValue
+            }
+        }
+    }
+  
+    nonisolated static func dependencies(
+        of keyPath: PartialKeyPath<Self>
+    ) -> TrackedProperties<Self> {
+        [keyPath]
+    }
+    
+    func purchase(alternateIcons: Bool, allRecipes: Bool) {
+        alternateIconsUnlocked = alternateIcons
+        allRecipesUnlocked = allRecipes
+    }
+}
+```
+
+When a property does not have a default value, that corresponding argument in the initializer does not have a default value. This means that the following example has a macro synthesized initializer of `init(a: Int, b: Int = 3)`.
+
+```swift
+@Observable final class InitializationSample {
+    var a: Int
+    var b: Int = 3
+}
+```
+
+Because the memberwise initializer and backing storage are generated together, the initializer is able to initialize that storage. User-defined initializers should call the generated memberwise initializer instead of attempting to initialize the properties directly.
+
+Due to limitations with macros, all fields must have type information; this restriction may be able to be lifted later when the type system grows a mechanism to detect inferred types.
```

Comment:
```suggestion
Since macros are only syntactic transformations, all fields must have explicit type information. This restriction may be able to be lifted later when the type system grows a mechanism to detect inferred types.
```